lib/kube/proxy/server.go

```diff
@@ -201,7 +201,7 @@ func (t *TLSServer) GetConfigForClient(info *tls.ClientHelloInfo) (*tls.Config,
 
 // GetServerInfo returns a services.Server object for heartbeats (aka
 // presence).
-func (t *TLSServer) GetServerInfo() (services.Server, error) {
+func (t *TLSServer) GetServerInfo() (services.Resource, error) {
```

Comment:
Can you clarify this change?

Reply:
Yeah, all existing Teleport services ("node", "app server" and "k8s service") are a part of the `services.Server` resource and the existing heartbeat code only supports announcing resources of type `services.Server`.

For database access I made a new resource type `services.DatabaseServer` and updated the heartbeat code to be more generic and support announcing of any resource so I could use it for a new DatabaseServer.

Application servers and k8s services will also need to become separate resources in future (e.g. `services.AppServer` and `sevices.KubeServer`) so they'll take advantage of this change as well.